includes/admin/class-wc-payments-admin.php

```diff
@@ -205,24 +218,40 @@ public function register_payments_scripts() {
 			true
 		);
 
-		$tos_script_src_url    = plugins_url( 'dist/tos.js', WCPAY_PLUGIN_FILE );
-		$tos_script_asset_path = WCPAY_ABSPATH . 'dist/tos.asset.php';
-		$tos_script_asset      = file_exists( $tos_script_asset_path ) ? require_once $tos_script_asset_path : [ 'dependencies' => [] ];
+		// phpcs:ignore WordPress.Security.NonceVerification
+		$tos_disabled_snackbar  = isset( $_GET['tos-disabled'] );
+		$tos_agreement_required = $this->is_tos_agreement_required();
+
+		if ( $tos_agreement_required || $tos_disabled_snackbar ) {
```

Comment:
Reminder: we shouldn't conditionally register scripts as discussed in slack (p1604409371181300-slack-CGGCLBN58).

Reply:
As mentioned in another comment as well, the script is now always registered, but conditionally enqueued and localized: https://github.com/Automattic/woocommerce-payments/pull/993/commits/8ad28b977f3e8bb150397473aadadaf845edfb29#diff-3863b4f88d0515e080fadc56af3c5e38e942afb96e569cb000ae6a594e8bea4aR289